### ISSUE TITLE
False Positive UnnecessaryApply

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1517]
## Expected Behavior
Nothing reported

## Observed Behavior
UnnecessaryApply reported

## Steps to Reproduce
```kt
list.add(
    if (something) {
        SomeObject().apply {
            x = y
        }
    } else {
        SomeOtherObject()
    }
)
```

The `apply` is necessary because the `SomeObject` instance is being used

## Your Environment
* Version of detekt used: RC14

**Comment by arturbosch** [comment]
Are you sure this is not fixed in RC14 ? I can't reproduce it.
The only statement inside the apply is an assignment which is handled by:
```
return expr.notAnAssignment() && expr.isMemberAccess()
```
